presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTester.java

```diff
@@ -116,19 +117,25 @@ public ConnectorId getCurrentConnectorId()
         public Optional<PlanNode> apply(PlanNode node, Captures captures, Context context)
         {
             PlanNodeMatcher planNodeMatcher = new PlanNodeMatcher(context.getLookup());
-            Set<Rule> matching = ruleSet.rules().stream()
-                    .filter(rule -> matches(rule.getPattern(), node, planNodeMatcher))
-                    .collect(toSet());
+            Map<Rule, Match<?>> matching = ruleSet.rules().stream()
```

Comment:
What is the benefit of testing that only one rule matches from a rule set?

Reply:
This is enforced by the convention used in tests. By things like `assertThat(...).on(...).matches()`, in this example `matches` expects that there were only single rule applied. So far that minimalistic approach was enough, This check is making sure that this assumption is still valid.